zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -461,9 +461,10 @@ protected boolean jobAbort() {
     if (interpreter == null) {
       return true;
     }
-    Scheduler scheduler = interpreter.getScheduler();
-    if (scheduler == null) {
-      return true;
+    try {
+      interpreter.cancel(getInterpreterContext(null));
+    } catch (InterpreterException e) {
+      throw new RuntimeException(e);
```

Comment:
why can't we throw InterpreterException here?

Reply:
This is due to signature of class `Job`, change the signature of class `Job` will result in many changes in its callers (I will do this kind of refactoring later),  so I just throw RuntimeException here, but the exception would still be caught by ZeppelinSever.